app/models/user.rb

```diff
@@ -110,4 +113,19 @@ def complete_registration
       registration.organization.update_attribute :active, true if registration.organization.present?
     end
   end
+
+  def technical_user_is_unique_in_org
+    # This validation make the assumption a technical user has only and only one membership at the creation
+    if User.joins(:organizations).where(name: name, 'organizations.id': memberships.first.organization.id).count > 0
+      errors.add :name, :must_be_unique_by_org
+    end
+  end
+
+  def normalize_name
+    self.normalized_name = if self.human?
+      "human-#{name.downcase}"
+    else
+      "technical-#{memberships.first.organization.normalized_name}-#{name.downcase}"
```

Comment:
This implementation may cause collisions in some cases. I created a separate issue for this: https://probe.storiesonboard.com/m/probe-dock-roadmap/!card/curfjHen8bIdbDnXJqDZ2w